src/NuGet.Core/NuGet.Resolver/PackageResolver.cs

```diff
@@ -131,7 +131,12 @@ public IEnumerable<PackageIdentity> Resolve(PackageResolverContext context, Canc
 
                 if (nonAbsentCandidates.Any())
                 {
-                    var circularReferences = ResolverUtility.FindCircularDependency(solution);
+                    //topologicallly sort non absent packages
+                    var sortedSolution = ResolverUtility.TopologicalSort(nonAbsentCandidates);
```

Comment:
I would expect this sort to be done inside ResolverUtility.FindCircularDependency, I don't see it being used anywhere outside of that method.